### PR TITLE
Adds Grid component

### PR DESCRIPTION
**Pull request by ShawnToubeau**
Grid component takes 1 prop `gridSize` represented as `{x: <value>, y: <value>}`

Screen Shot 2020-02-18 at 3 44 20 PM

### CODE REVIEW

File: src/views/index.tsx [hunk 1]
```diff
@@ -1,19 +1,13 @@
-import * as React from "react";
```
Comment (uptonm) [inline src/views/index.tsx]: We should probably keep the Root view as simple as possible, abstracting as much component logic to child components so we have an end result like.
```
<div className="root">
    <Navbar />
    <Canvas />
</div>
```

File: src/components/Grid/Grid.scss [hunk 2]
```diff
@@ -0,0 +1,3 @@
+.Grid {
+  height: 100vh !important;
```
Comment (uptonm) [inline src/components/Grid/Grid.scss]: This height will need to change down the line as the Menu Bar will take up some of that height. We might just need to set `.root` to 100vh, and have the canvas fill up the remainder of the space

File: src/views/index.tsx (hunk 1 above)
Comment (ShawnToubeau) [reply src/views/index.tsx]: Agreed. I was using that wrapper around the Grid for debugging. Added it as a temp style to the `Grid.scss`

**Comment by ShawnToubeau** [comment]
@JamesCoyne is handling the meshes for the smaller `Space` components. The stuff I have is just temp so that I could get the `Grid` component working. The idea is one his components are done it'll be easy to swap them out

**Comment by ShawnToubeau** [comment]
> We should move Canvas to its own component, and that component would contain all of the lights/meshes/etc.

Can you explain your reasoning behind this? From the way I view it the Canvas (and everything inside of it like `ambientLight` and `pointLight`) are properties of our grid and I think that it's minimal to the point where we can keep it in the `Grid` component where if we extracted it out to another file it would get confusing because it's another file to keep track of

**Comment by uptonm** [comment]
@all-contributors-bot add @ShawnToubeau for code, infrastructure, design, docs

**Comment by uptonm** [comment]
@all-contributors add @ShawnToubeau for code, infrastructure, design, docs

**Pull request merged**